duolingo.py

```diff
@@ -59,17 +67,43 @@ def _login(self):
         """
         Authenticate through ``https://www.duolingo.com/login``.
         """
+        jwt = self._load_session_from_file()
+        if jwt:
+            self.jwt = jwt
+            logged_in = self._check_login()
+            if logged_in:
+                return True
+            else:
```

Comment:
This `else` branch is not required, after `return` statement, it won't be reachable anyway

Reply:
Unless the token in the file has expired, or is otherwise incorrect. Then the login check will fail, it'll come into this `else` block, and then reset the Duolingo.jwt to None before doing a new login

Reply:
Sorry, I mean not the whole branch, just the `else` statement is redundant. 
```python
if self._check_login():
    return True
self.jwt = None
```
Actually variable is also redundant since it's not used later in the code and the return value can be used directly. Or even:

```python
self.jwt = self._load_session_from_file()
if self._check_login():
    return True
self.jwt = None
```

Why doesn't `_load_session_from_file` write to self.jwt directly? Like `_save_session_to_file` saves from it, for consistency. Then it can be reduced to something like this:
```python
if self._load_session_from_file() and self._check_login():
    return true
self.jwt = None
```

Reply:
How about now, with the jwt argument, I've neatened it up a bit.

I'm only reading jwt from file if it's None, such that you can specify a start jwt to use, and then it'll be saved to file to be updated in the future